src/main/java/frc/robot/commands/arm/JiggleNote.java

```diff
@@ -23,21 +23,22 @@ public void initialize() {
 
     @Override
     public void execute() {
+
+    }
+
+    @Override
+    public boolean isFinished() {
         var sensor = !shooterTrack.getSensor();
-        if (sensor) {
+        if (!shooterTrack.getSensor() && Timer.getFPGATimestamp() - startTime > length) {
```

Comment:
you store `shooterTrack.getSensor()` above in `sensor` but grab it again in the if statement; just use the variable. also if you're using static types then I wouldn't suggest using `var` also (that being said, there's no "best practice" on this just personal preference for readability without compile).